common/logisticspipes/proxy/recipeproviders/LogisticsCraftingTable.java

```diff
@@ -31,19 +32,15 @@ public boolean importRecipe(TileEntity tile, IItemIdentifierInventory inventory)
 			return false;
 		}
 
+		// Craft result is slot 9
 		inventory.setInventorySlotContents(9, result);
 
-		// Import
-		for (int i = 0; i < bench.matrix.getSizeInventory(); i++) {
-			if (i >= inventory.getSizeInventory() - 2) {
-				break;
-			}
-			ItemStack stackInSlot = bench.matrix.getStackInSlot(i);
-			if (!stackInSlot.isEmpty() && stackInSlot.getCount() > 1) {
-				stackInSlot = stackInSlot.copy();
-				stackInSlot.setCount(1);
+		// Import (Slots [0, 8])
+		for (Pair<ItemIdentifierStack, Integer> slot : bench.matrix) {
+			final int i = slot.getValue2();
+			if (i <= inventory.getSizeInventory() - 2) {
```

Comment:
not exactly sure why `inventory size - 2` would be okay here in the new code, but not previously? Can you elaborate?